test/NuGet.Core.Tests/NuGet.Configuration.Test/SettingsTestUtils.cs

```diff
@@ -37,6 +38,34 @@ public static string RemoveWhitespace(string s)
             return Regex.Replace(s, @"\s+", string.Empty);
         }
 
+        public static bool SequenceDeepEquals<T>(IList<T> settings1, IList<T> settings2) where T : SettingBase
+        {
+            if (settings1 == null && settings2 == null)
+            {
+                return true;
+            }
+
+            if (settings1 == null || settings2 == null)
+            {
+                return false;
+            }
+
+            if (settings1.Count() != settings2.Count())
+            {
+                return false;
+            }
+
+            for (var i = 0; i < settings1.Count(); i++)
+            {
+                if (!DeepEquals(settings1[0], settings2[0]))
```

Comment:
First, you should be passing `i` instead of a hard-coded `0`.

Second, is order really significant?  The code assumes order is significant, but I'm not convinced it should be.

Reply:
I originally wrote it assuming the order was significant since it was the easiest implementation, but I can change it to not assume an order and order them before comparing anything.